crates/api_crud/src/comment/create.rs

```diff
@@ -97,16 +98,18 @@ pub async fn create_comment(
   )
   .await?;
 
-  let comment_form = CommentInsertForm {
+  let mut comment_form = CommentInsertForm {
     language_id: Some(language_id),
     ..CommentInsertForm::new(local_user_view.person.id, data.post_id, content.clone())
   };
+  plugin_hook_mut("create_local_comment", &mut comment_form)?;
```

Comment:
Shouldnt these be named like `pre` and `post` create hooks? 

Create and New are confusing.

Reply:
Also I'd highly recommend these being proper isolated functions. Have them take in and return forms, not mutate them. Otherwise its difficult to try to figure out state changes.

Reply:
For the hook names its easier to discuss in the RFC which has them all in one place. I dont mind changing them.

https://github.com/LemmyNet/rfcs/pull/8/files#diff-3026a7e22ac85145e76c9e7a03eb52302818fb60c05ba10b98895c9f5768999cR44

The pre hook could be either of these, its practically the same. In both cases the plugin can change all fields (thats the purpose), using a mut param is just shorter to write.
```rust
plugin_hook_mut("create_local_comment", &mut comment_form)?;
comment_form = plugin_hook_mut("create_local_comment", &comment_form)?;
```

Reply:
Updated plugin hooks with clearer names (before, after), and added some missing hooks after post update etc.